[pyqgis] add a code snippet specific for plugins (#5973)
qgis/QGIS-Documentation: docs/pyqgis_developer_cookbook/plugins/snippets.rst

        QgsProject,
    )

    from qgis.gui import (
        QgsOptionsWidgetFactory,
        QgsOptionsPageWidget,
        QgsMapLayerConfigWidgetFactory,

>         QgsMapLayerConfigWidget

Reviewer: Do you think we need to import these if we don't use them?

Author: uh that's right! I forgot to remove them after some experiment ;)

    # The method requires one parameter (the new value for the second
    # field of the selected feature(s)) and can be called by
    change_value(50)

.. index:: Plugins; Customization

Interface for plugin in the options dialog
------------------------------------------

> You can add a custom plugin options tab to :menuselection:`Settings --> Options`. The 

Reviewer 2: ```suggestion
You can add a custom plugin options tab to :menuselection:`Settings --> Options`.
This is preferable over adding a specific main menu entry for your plugin's options, as it keeps all of the 
QGIS application settings and plugin settings in a single place which is easy for user's to discover and navigate.
  The 
```

> following snippet will just add the plugin tab without any option accessible.

Reviewer 2: ```suggestion
following snippet will just add a new blank tab for the plugin's settings, ready for you to populate with all the
options and settings specific to your plugin.
```

You can split the following classes into different files. In this example, we are
adding two classes into the main :file:`mainPlugin.py` file.

.. testcode:: plugin_snippets

    class MyPluginOptionsFactory(QgsOptionsWidgetFactory):

        def __init__(self):

>             super(QgsOptionsWidgetFactory, self).__init__()

Reviewer 2: ```suggestion
            super().__init__()
```

        def icon(self):

>             return QgsApplication.getThemeIcon('icon.svg')

Reviewer 2: Generally you won't be using QgsApplication.getThemeIcon for plugins, because that's only usable for icons shipped with QGIS itself. Maybe better to just do something simple like

    return QIcon('icons/my_plugin_icon.svg')

        def createWidget(self, parent):
            return ConfigOptionsPage(parent)

    class ConfigOptionsPage(QgsOptionsPageWidget):

        def __init__(self, parent):

>             super(ConfigOptionsPage, self).__init__(parent)

Reviewer 2: ```suggestion
            super().__init__(parent)
```

            layout = QHBoxLayout()
            layout.setContentsMargins(0, 0, 0, 0)
            self.setLayout(layout)

Finally we are adding the imports and modifying the ``__init__`` function:

.. testcode:: plugin_snippets

    from qgis.PyQt.QtWidgets import QHBoxLayout
    from qgis.gui import QgsOptionsWidgetFactory, QgsOptionsPageWidget

    class MyPlugin:
        """QGIS Plugin Implementation."""

        def __init__(self, iface):
            """Constructor.

            :param iface: An interface instance that will be passed to this class
                which provides the hook by which you can manipulate the QGIS
                application at run time.
            :type iface: QgsInterface
            """
            # Save reference to the QGIS interface
            self.iface = iface

>             self.options_factory = MyPluginOptionsFactory()
>             self.options_factory.setTitle(self.tr('My Plugin'))
>             iface.registerOptionsWidgetFactory(self.options_factory)

Reviewer 2: ```suggestion

        def initGui(self):
            self.options_factory = MyPluginOptionsFactory()
            self.options_factory.setTitle(self.tr('My Plugin'))
            iface.registerOptionsWidgetFactory(self.options_factory)

        def unload(self):
            iface.unregisterOptionsWidgetFactory(self.options_factory)
```

    # The method requires one parameter (the new value for the second
    # field of the selected feature(s)) and can be called by
    change_value(50)

.. index:: Plugins; Customization

Interface for plugin in the options dialog
------------------------------------------

You can add a custom plugin options tab to :menuselection:`Settings --> Options`.
This is preferable over adding a specific main menu entry for your plugin's 
options, as it keeps all of the QGIS application settings and plugin settings in 

> a single place which is easy for user's to discover and navigate.

Reviewer: ```suggestion
a single place which is easy for users to discover and navigate.
```

> The following snippet will just add a new blank tab for the plugin's settings, 
> ready for you to populate with all the options and settings specific to your 
> plugin.
> 
> You can split the following classes into different files. In this example, we are
> adding two classes into the main :file:`mainPlugin.py` file.

Reviewer: Have a separate paragraph for what is actually about our example and the general purpose of the feature?
```suggestion

The following snippet will just add a new blank tab for the plugin, 
ready for you to populate with all the specific options and settings.
You can split the following classes into different files. In this example, we are
adding two classes into the main :file:`mainPlugin.py` file.
```
